lib/internal/fs/recursive_watch.js

```diff
@@ -202,6 +202,9 @@ class FSWatcher extends EventEmitter {
         this.emit('change', 'rename', pathRelative(this.#rootPath, file));
       } else if (currentStats.isDirectory()) {
         this.#watchFolder(file);
+      } else {
+        // Watching a dircetory will trigger a change event for child files
+        this.emit('change', 'change', pathRelative(this.#rootPath, file));
```

Comment:
If your comment is true, than line 203 `currentStats.isDirectory` check will be executed. Since this is a `watchFile` function. I believe you wanted to write: "Watching a file should trigger changes if the file is changed"

Reply:
`currentStats.isDirectory` refers to the changed object not to the watched object.

Reply:
On macOS, this line only fires `rename` event. 

```
{ event: 'rename', filename: 'file-1.txt' }
```

In Linux this line fires:

```
{ event: 'change', filename: 'file-1.txt' }
```

Reply:
You need to change `change` event to `rename`

Reply:
it is inconsistent. if you try multiple times with and without timeouts you will get different results

Reply:
It behaves differently with and without timeout because of the internals of the `recursive_watch` file. For this particular change, this line needs to be:

```suggestion
        this.emit('change', 'rename', pathRelative(this.#rootPath, file));
```

Reply:
why? did a rename occur?

Reply:
I'm just imitating the existing behavior on macOS platform. On macOS it returns `rename` event. Whether or not it this is correct or not is a different discussion (possibly in `libuv` repo).